cats/cats-test/src/main/groovy/com/netflix/spinnaker/cats/provider/ProviderCacheSpec.groovy

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2014 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.cats.provider
+
+import com.netflix.spinnaker.cats.agent.CacheResult
+import com.netflix.spinnaker.cats.agent.DefaultCacheResult
+import com.netflix.spinnaker.cats.cache.*
+import com.netflix.spinnaker.cats.mem.InMemoryCache
+import com.netflix.spinnaker.cats.provider.DefaultProviderCache
+
+abstract class ProviderCacheSpec extends CacheSpec {
+
+    WriteableCache backingStore
+
+    @Override
+    Cache getSubject() {
+        backingStore = new InMemoryCache()
+        new DefaultProviderCache(backingStore)
+    }
+
+    void populateOne(String type, String id, CacheData cacheData = createData(id)) {
+        defaultProviderCache.putCacheResult('testAgent', [], new DefaultCacheResult((type): [cacheData]))
+    }
+
+    ProviderCache getDefaultProviderCache() {
+        getCache() as DefaultProviderCache
+    }
+
+    def 'explicit evictions are removed from the cache'() {
+        setup:
+        String agent = 'agent'
+        CacheResult result = new DefaultCacheResult(test: [new DefaultCacheData('id', [id: 'id'], [:])])
+        defaultProviderCache.putCacheResult(agent, [], result)
+
+        when:
+        def data = defaultProviderCache.get('test', 'id')
+
+        then:
+        data != null
+        data.id == 'id'
+
+        when:
+        defaultProviderCache.putCacheResult(agent, [], new DefaultCacheResult([:], [test: ['id']]))
+        data = defaultProviderCache.get('test', 'id')
+
+        then:
+        data == null
+    }
+
+    def 'multiple agents can cache the same data type'() {
+        setup:
+        String usEast1Agent = 'AwsProvider:test/us-east-1/ClusterCachingAgent'
+        CacheResult testUsEast1 = buildCacheResult('test', 'us-east-1')
+        String usWest2Agent = 'AwsProvider:test/us-west-2/ClusterCachingAgent'
+        CacheResult testUsWest2 = buildCacheResult('test', 'us-west-2')
+        defaultProviderCache.putCacheResult(usEast1Agent, ['serverGroup', 'cluster', 'application'], testUsEast1)
```

Comment:
This PR also adds `cluster` and `application` to the List of authoritative types when storing test data. This is for use in a prototype `ProviderCache` that only stores full cache records for Authoritative types. In this implementation, Informative types only update the relationships graph.